modules/monitoring-creating-cluster-monitoring-configmap.adoc

```diff
@@ -3,36 +3,55 @@
 // * monitoring/cluster_monitoring/configuring-the-monitoring-stack.adoc
 
 [id="creating-cluster-monitoring-configmap_{context}"]
-= Creating cluster monitoring ConfigMap
+= Creating a cluster monitoring ConfigMap
 
-To configure the Prometheus Cluster Monitoring stack, you must create the cluster monitoring ConfigMap.
+To configure the {product-title} monitoring stack, you must create a cluster monitoring ConfigMap.
 
 .Prerequisites
 
-* An installed `oc` CLI tool
-* Administrative privileges for the cluster
+* You have access to the cluster as a user with the cluster-admin role.
+* You have installed the OpenShift CLI (`oc`).
 
 .Procedure
 
-. Check whether the `cluster-monitoring-config` ConfigMap object exists:
+. Check whether a `cluster-monitoring-config` ConfigMap object exists:
 +
+[source,terminal]
 ----
 $ oc -n openshift-monitoring get configmap cluster-monitoring-config
 ----
 
-. If it does not exist, create it:
+. Configure the ConfigMap.
+** If the ConfigMap does not exist:
```

Comment:
I feel the original is good and would like to suggest it. 
The first item is "Check whether the cluster-monitoring-config ConfigMap object exists:", then the second item should be  "If it does not exist, create it:".
The third item is "editing the cluster-monitoring-config ConfigMap:"
Move "Create the `data` section if it does not exist yet:" as sub item of the third item

Reply:
@lihongan thank you for the suggestion! I tried a number of different combinations in [my related PR](https://github.com/openshift/openshift-docs/pull/24722) which also updates this section. In my view, 'Editing the cluster-monitoring-config ConfigMap' is something that is not successive to creating it. Instead, it only needs to be done if it already exists. So, the steps are not sequential, but alternative. In the procedure, we create the data section at the same time as create a new ConfigMap if it does not exist. Then, if it already does, we just check to make sure that the data section is already there. This is an either or in my opinion, not a sequence.

Reply:
@pneedle-rh agree with you, comparing to currently "Configure the ConfigMap" is successive to "Check whether a cluster-monitoring-config ConfigMap object exists:", feel that it's more better  'Editing the cluster-monitoring-config ConfigMap' is something that is not successive to creating it. Maybe can just remove "Configure the ConfigMap", move "If it does not exist, create it:" to an upper level

Reply:
@thank you for your response! IF we remove the Configure the ConfigMap section, how would we ensure that the `data/config.yaml` section is in place in the case where the ConfigMap already exists, in preparation for the subsequent modules which rely on that being there?

Reply:
Essentially this module is teeing up a prerequisite and an .Additional resources reference in subsequent sections in the 'Configuring the monitoring stack' assembly, as seen in my related 4.6 PR here: https://osdocs1240_user_workload_monitoring--ocpdocs.netlify.app/openshift-enterprise/latest/monitoring/configuring-the-monitoring-stack.html

Reply:
> @lihongan thank you for your response! If we remove the Configure the ConfigMap bullet point, how would we ensure that the `data/config.yaml` section is in place in the case where the ConfigMap already exists, in preparation for the subsequent modules which rely on that being there?

I mean just remove item title "Configure the ConfigMap", it should be like this
1. Check whether the cluster-monitoring-config ConfigMap object exists:
2. If the ConfigMap does not exist:
3. If the ConfigMap already exists:

Reply:
I agree with removing the item title "Configure the ConfigMap". This module is for *creating* the configmap object, configuring it in a meaningful way is the next step, outside of this module. 
Basically what we want is to edit the second step from original documentation https://docs.openshift.com/container-platform/4.4/monitoring/cluster_monitoring/configuring-the-monitoring-stack.html#creating-cluster-monitoring-configmap_configuring-monitoring in such way that instead of using oc create, it instructs to apply yaml manifest.
It would be much easier if we could just add something like `--from-literal=config.yaml='|'`, but unfortunately the resulting cm always has the pipe sign wrapped in quotes and it breaks the operator in the same way.

Reply:
@lihongan @olga-mir these two items are not sequential, they are either or:

* If the ConfigMap does not exist:
* If the ConfigMap already exists:

As such, I really think that they should be bullet points in a single step, which is why I am suggesting that. Perhaps we can replace the phrase 'Configure the ConfigMap' with something else if that doesn't sit well with you? I don't think they should be sequential steps though, as they are mutually exclusive.

Reply:
looking through this rendered document, this approach creates too much indentation. https://osdocs1240_user_workload_monitoring--ocpdocs.netlify.app/openshift-enterprise/latest/monitoring/configuring-the-monitoring-stack.html

This section is focusing on creating the config map. There is nothing much to the "editing" part of it. Especially now that "oc create" is replaced with applying the manifest, it creates unnecessary duplication. I feel like this module is way too much wording for not much value.

Reply:
@lihongan @olga-mir please see this preview in my related assembly. This provides a preview example of the flow of the steps I am recommending:

https://osdocs1240_user_workload_monitoring--ocpdocs.netlify.app/openshift-enterprise/latest/monitoring/configuring-the-monitoring-stack.html#creating-cluster-monitoring-configmap_configuring-the-monitoring-stack

Reply:
Hi, I have to say you should @lihongyan1 not @lihongan :)

Reply:
Ah my apologies! Thank you for highlighting that!

Reply:
> looking through this rendered document, this approach creates too much indentation. https://osdocs1240_user_workload_monitoring--ocpdocs.netlify.app/openshift-enterprise/latest/monitoring/configuring-the-monitoring-stack.html
> 
> This section is focusing on creating the config map. There is nothing much to the "editing" part of it. Especially now that "oc create" is replaced with applying the manifest, it creates unnecessary duplication. I feel like this module is way too much wording for not much value.

@olga-mir @lihongyan1 thank you for your responses. Olga your comment above makes a good point, because all of the subsequent configuration modules include the necessary data/config.yaml components. How about this solution?:

  https://github.com/openshift/openshift-docs/pull/24722/files#diff-89302581129a792feee2baace787a6db

If that is suitable, please apply something similar to this PR. Thank you!